sqlmesh/core/renderer.py

```diff
@@ -293,6 +295,14 @@ def _cache_key(
             runtime_stage,
         )
 
+    @contextmanager
+    def _normalize_and_quote(self, query: E) -> t.Iterator[E]:
```

Comment:
why don't we use the method in dialect now? add the quote_identifiers flag to that one and use it?

Reply:
That's what I did initially, but then I noticed that we only care about not quoting in this one case and I concluded that the difference in logic is sufficient enough to just re-implement it here. Do you suggest to share the logic anyway?

Reply:
yea, i think so

Reply:
fixed